### PR TITLE
Faster RCNN Model + Pascal VOC DataModule

### PR DESCRIPTION
**Pull request by teddykoker**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [X] Did you make sure to update the docs?   
- [X] Did you write any new necessary tests?  

## What does this PR do?

1. Implements the Pascal VOC Detection dataset as a `DataModule`
2. Implements Faster RCNN as a `LightningModule`

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pl_bolts/datamodules/vocdetection_datamodule.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
+import torch
```
Comment (Borda) [inline pl_bolts/datamodules/vocdetection_datamodule.py:1]: no need to repeat `_datamodule.py` in the name

File: pl_bolts/models/detection/faster_rcnn.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        parser.add_argument("--pretrained_backbone", type=bool, default=True)
+        parser.add_argument("--trainable_backbone_layers", type=int, default=3)
+        parser.add_argument("--replace_head", type=bool, default=True)
+
+        parser.add_argument("--data_dir", type=str, default=".")
+        parser.add_argument("--batch_size", type=int, default=1)
+        return parser
+
+
+def cli_main():
```
Comment (Borda) [inline pl_bolts/models/detection/faster_rcnn.py:129]:
```suggestion
def run_cli():
```
let's add test for the script too

File: pl_bolts/datamodules/vocdetection_datamodule.py (hunk 1 above)
Comment (teddykoker) [reply pl_bolts/datamodules/vocdetection_datamodule.py:1]: I figured I would name it to be consistent with `imagenet_datamodule.py`, `mnist_datamodule.py`, etc. We may just want to rename all of these in the future.

File: pl_bolts/models/detection/faster_rcnn.py (hunk 2 above)
Comment (teddykoker) [reply pl_bolts/models/detection/faster_rcnn.py:129]: Question about tests, in this case the cli needs to download the Pascal VOC data, which is 2GB and takes some time, should I still implement?

**Comment by teddykoker** [comment]
@Borda didn't add test for cli, but added test training with `DummyDetectionDataset`, which I just added.

File: pl_bolts/datamodules/vocdetection_datamodule.py (hunk 1 above)
Comment (williamFalcon) [reply pl_bolts/datamodules/vocdetection_datamodule.py:1]: i agree with consistency and yes, we need to keep the _datamodule.py name

File: tests/models/test_detection.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    image = torch.rand(1, 3, 400, 400)
+    model(image)
+
+
+def test_fasterrcnn_train(tmpdir):
+
+    model = FasterRCNN()
+
+    train_dl = DataLoader(DummyDetectionDataset(), collate_fn=_collate_fn)
```
Comment (williamFalcon) [inline tests/models/test_detection.py:25]: We have a dummy dataset already. why is this one different?

Comment (teddykoker) [reply tests/models/test_detection.py:25]: Instead of a tensor label it returns a dict with values `{"boxes": FloatTensor[N, 4], "labels": Int64Tensor[N]}` as per the standard in pytorch for detection models. Should this be a choice in the existing dummy dataset?

**Comment by teddykoker** [comment]
Also `VOCDetectionDataModule` won't work right now as they modified an image in the dataset that effects the md5sum of the download. I added a fix to torchvision thats been approved: https://github.com/pytorch/vision/pull/2589

File: pl_bolts/datamodules/vocdetection_datamodule.py (hunk 1 above)
Comment (Borda) [reply pl_bolts/datamodules/vocdetection_datamodule.py:1]: well, let's talk about refactoring later...
